### PR TITLE
refactor: simplify challenges when answer is fixed

### PR DESCRIPTION
**Pull request by nbaars**
When the answer is always the same we can cache it and the challenge only need to implement one method. This removes the necessity to implement methods `spoiler()` and `answerCorrect()` in every subclass.

### CODE REVIEW

**Comment by commjoen** [comment]
deploying the PR to https://arcane-scrubland-42646.herokuapp.com/

File: src/main/java/org/owasp/wrongsecrets/challenges/FixedAnswerChallenge.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * <p>Why do we make this distinction? Because in the case of the fixed answer we can cache the
+ * value. It is important to <b>NOT</b> do any reading / calculation in the constructor when using
+ * this interface.
+ *
+ * <p>NOTE: If the challenge depends on a calculation you can implement {@link Challenge}
+ */
+public abstract class FixedAnswerChallenge implements Challenge {
+
+  private Supplier<String> cachedAnswer = Suppliers.memoize(() -> getAnswer());
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/FixedAnswerChallenge.java:19]: CAn this be renamed to memorize?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -275,64 +275,52 @@ These are the things that you have to keep in mind.
...
       @Test
       void rightAnswerShouldSolveChallenge() {
-          var challenge = new Challenge28(scoreCard);
+          var challenge = new Challenge28();
           Assertions.assertThat(challenge.solved("wrong answer")).isFalse();
           Assertions.assertThat(challenge.solved(challenge.spoiler().solution())).isTrue();
       }
     }
 ```
+
```
Comment (commjoen) [inline CONTRIBUTING.md:323]: Can we add an example of a challenge with a random answer as well please?

**Review by commjoen — CHANGES_REQUESTED** [review]
after doing some testing and reviewing: only a few small comments: can you have a look please?

**Review by bendehaan — APPROVED** [review]

**Pull request merged**